### ISSUE TITLE
Saving to different file names

### ISSUE THREAD

**Issue by katsanz13** [AlexeyAB/darknet#125]
I have compiled darknet with cuda 8.0 but not open cv. I have successfully trained yolo for multi-object detection. I am attempting to input a text file, with a list of images for detection. By default, whenever you test yolo the detection predictions is outputted to a png named predictions. I would like to name all the files by their file name. Does anyone know how to do this?

**Comment by AlexeyAB** [comment]
Do you use Linux?

1. You can try to use command: `./darknet detector valid data/voc.data yolo-voc.cfg yolo-voc.weights` - this will save prediction results to the .txt-files in the directory `/results`

    Before this:
    * Create directory `results` near with file `darknet`
    * Change the line `valid  = data/my_images_list.txt` int the file `voc.data`
    * Also find function `validate_detector` and check that there is such line `char *valid_images = option_find_str(options, "valid", "data/train.list");`  https://github.com/AlexeyAB/darknet/blob/5bc62b14e06a3fcfda4e3a19fba77589920eddee/src/detector.c#L238

    * If you add line `eval=coco` at the end of `voc.data`, then prediction results will be saved to the JSON instead of txt-files: https://github.com/AlexeyAB/darknet/blob/5bc62b14e06a3fcfda4e3a19fba77589920eddee/src/detector.c#L268

----

2. Or try to use bash-file, something look like this - put it to `batch.sh` file and run it:

```
while read line
do
echo $line
./darknet detector test data/voc.data yolo-voc.cfg yolo-voc.weights -i 0 $line
cp -R ./predictions.png ./$line.png
done < "/root/test_images.txt"
```

3. Or change the source code to do this )

**Comment by AlexeyAB** [comment]
@katsanz13 Yes, you can do it simply on Windows or Linux by using this repo, just:

* uncomment these 2 lines: `draw_boxes` and `cv::imwrite` https://github.com/AlexeyAB/darknet/blob/9c1b9a2cf6363546c152251be578a21f3c3caec6/src/yolo_console_dll.cpp#L103
* comment these 2 lines: `cv::imshow` and `cv::waitKey` https://github.com/AlexeyAB/darknet/blob/9c1b9a2cf6363546c152251be578a21f3c3caec6/src/yolo_console_dll.cpp#L34

And then: 

On Windows: 
* at first compile `build\darknet\yolo_cpp_dll.sln` and then compile `build\darknet\yolo_console_dll.sln`
* run `yolo_console_dll.exe`
* enter txt file name with list of images, for example, `air.txt` - extension `txt` must be mandatory - you will see `res_....` - image files with bounded boxes

----

On Linux: 
* change line to `LIBSO=1` in the Makefile: https://github.com/AlexeyAB/darknet/blob/master/Makefile#L6
* type: `make` to build darknet.so and `uselib`-runable file
* run `uselib`
* enter txt file name with list of images, for example, `air.txt` - extension `txt` must be mandatory - you will see `res_....` - image files with bounded boxes

----

Example of input `air.txt`:

> air1.jpg
> air2.jpg
> air3.jpg
> air4.jpg

**Comment by GianniCristian** [comment]
Dear All, 

Thanks to you, I have managed in running 'uselib' correctly. I would like to know how it is possible to set the threshold. Is it also possible to save the output results on a txt file?

Thanks

**Comment by AlexeyAB** [comment]
@GianniCristian Hi,
You can just 
* put filenames of images to the text file `air.txt`, 1 filename per line
* run: `uselib.exe air.txt > result.txt`

----

To change threshold to 0.3, change this line: https://github.com/AlexeyAB/darknet/blob/4d2fefd75a57dfd6e60680eaf7408c82e15a025d/src/yolo_console_dll.cpp#L117
to this: `std::vector<bbox_t> result_vec = detector.detect(mat_img, 0.3);`

**Comment by GianniCristian** [comment]
Thanks!

**Comment by zenandibarkah** [comment]
i want save the result of image to file txt. but i the result still not save in my folder. please give me solution for that. thanks

**Comment by zenandibarkah** [comment]
i want save the result of image to file txt. but the result still not save in my folder. please give me solution for that. thanks

**Comment by AlexeyAB** [comment]
@zenandibarkah What command do you use?

**Comment by zenandibarkah** [comment]
i solve that problem.. now i have a new problem. i want to show output in the terminal just the biggest number of object detected. 

example output:
car - 23 (it means 23 car detected)
truck - 2 (it means 2 truck detected)

i have done edited in **void**_show_result_ but im stuck :( 
thanks

**Comment by LynneHan** [comment]
@AlexeyAB so another problem is that i wanna calculate the number of bounding box for every images,how can i do this ?

**Comment by spaul13** [comment]
I have already made the changes @AlexeyAB to save the predictions into different files changing the Makefile and also modifying the cpp code and its generating uselib and libdarknet.so but while executing this command
 **./uselib air.txt > result.txt** it's showing the error message

**./uselib: error while loading shared libraries: libdarknet.so: cannot open shared object file: No such file or directory**

@AlexeyAB, @GianniCristian and others can anyone please tell me why I can't run the uselib properly?

**Comment by panca1** [comment]
@AlexeyAB How can I store multiple image results into different directory?
